### DISCUSSION TITLE
Objects without prototypes are causing errors in body request.

### DISCUSSION THREAD

**Discussion by gabrielferreiraa** [apollographql/apollo-server#5643] [Ideas]
Hi guys. Can you help me to understand if this behavior is a bug or not?

Here, is this line -> https://github.com/apollographql/apollo-server/blob/main/packages/apollo-datasource-rest/src/RESTDataSource.ts#L235

We have one validation to check if my body is `object` or an `array`. Right?

But, this object validation in my view, its wrong. Because, not all objects in Javascript has `Object` as constructor/prototype.

E.g: `Object.create` doesn't have a constructor -> http://adripofjavascript.com/blog/drips/creating-objects-without-prototypes.html

In some cases, when you get your body request payload and put directly in your body (`this.post('/url', data)`) we have some errors... to bypass this error we need construct an object with constructor, like a : `this.post('/url', { ...data }` 

To resolve this, in my fork, I modified [this line](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-datasource-rest/src/RESTDataSource.ts#L235) and wrote this

```javascript
const isObject = (value: unknown) => Object.prototype.toString.call(value) === '[object Object]';

if (
   options.body !== undefined &&
   options.body !== null &&
   (isObject(options.body) ||
      Array.isArray(options.body) ||
      ((options.body as any).toJSON &&
         typeof (options.body as any).toJSON === 'function'))
) {
   // ....
}
```

With that, we guarantee that any object will work in body request.

**Comment by glasser** [comment]
Is that the same as https://github.com/apollographql/apollo-server/pull/5070 ?
